Patch-Source: https://gitlab.alpinelinux.org/Cogitri/apk-polkit-rs/-/commit/940ac080f4a5da3fa06b9d041ee25e4b2beb9834
---
From 940ac080f4a5da3fa06b9d041ee25e4b2beb9834 Mon Sep 17 00:00:00 2001
From: Achill Gilgenast <achill@achill.org>
Date: Sun, 16 Nov 2025 14:54:26 +0100
Subject: [PATCH] data: Add systemd system service

Oriented at the postmarketOS service: https://gitlab.postmarketos.org/postmarketOS/pmaports/-/blob/master/extra-repos/systemd/systemd-services/system-apk-polkit-server.service

Allows packaging the service file in a $pkgname-systemd subpackage in
Alpine.
---
 data/apk-polkit-server.service.in | 10 ++++++++++
 data/meson.build                  |  7 +++++++
 2 files changed, 17 insertions(+)
 create mode 100644 data/apk-polkit-server.service.in

diff --git a/data/apk-polkit-server.service.in b/data/apk-polkit-server.service.in
new file mode 100644
index 0000000..391650c
--- /dev/null
+++ b/data/apk-polkit-server.service.in
@@ -0,0 +1,10 @@
+[Unit]
+Description=PolKit server to manage APK through DBus
+
+[Service]
+ExecStart=@LIBEXEC_DIR@/apk-polkit-server @EXTRA_ARGS@
+Restart=always
+RestartSec=1
+
+[Install]
+WantedBy=multi-user.target
diff --git a/data/meson.build b/data/meson.build
index 140b10c..849e227 100644
--- a/data/meson.build
+++ b/data/meson.build
@@ -67,6 +67,13 @@ configure_file(
     install: true,
     install_dir: get_option('sysconfdir') / 'init.d',
 )
+configure_file(
+    input: 'apk-polkit-server.service.in',
+    output: 'apk-polkit-server.service',
+    configuration: conf,
+    install: true,
+    install_dir: get_option('prefix') / 'lib' / 'systemd' / 'system',
+)
 
 metainfo_file = files('dev.Cogitri.apkPolkit2.metainfo.xml')[0]
 install_data(
-- 
GitLab

